print-rt6.c

```diff
@@ -33,13 +33,118 @@
 
 #include "ip6.h"
 
+static int
+srh_tlv_print(netdissect_options *ndo, const u_char *p, u_int bytes_left)
+{
+	u_int tlv_type, tlv_len;
+	uint8_t parse_next = 1;
+	while (parse_next)
+	{
+		tlv_type = GET_U_1(p);
+		p += 1;
+		bytes_left -= 1;
```

Comment:
Use ND_ICHECK() before decrement.

The recommended way is:
Test if we can't decrement length thus invalid, decrement length, increment pointer. like:
ND_ICHECKMSG_U("remaining length", bytes_left, <, 1);
bytes_left -= 1;
p += 1;

With, at the end of srh_tlv_print():
 
invalid:
   return -1;

Reminder for ND_ICHECK*() macros: If the checked expression is true an error message is printed and a 'goto invalid' is executed.

This should avoid incorrect writing: "(invalid TLV length ..., bytes left ...) (invalid)"

Reply:
I wasn't aware of that macro, thanks for pointing it out, I am now using it in the code